backend/src/utils/error.rs

```diff
@@ -108,7 +110,8 @@ impl error::ResponseError for Error {
             | Error::Request(_)
             | Error::FailedToSendEmail(_)
             | Error::InvalidJson(_)
-            | Error::FailedToBuildEmail(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            | Error::FailedToBuildEmail(_)
+            | Error::UrlParsingError(_) => StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
It doesn't match other errors, please use `InvalidUrl` or even better `CouldNotParseUrl` or somehting like that